### PR TITLE
Adds a function for computing the hash of any value

### LINKED ISSUE #43: Add a function for computing the hash any value
With the addition of a DOM-style API to ion-js, a convenience function should be added to compute the hash of any value (including those that are instances of DOM types as well as native-types).

Proposed signature:
```javascript
function digest(value: any, algorithm: string): Uint8Array;
```

example usage:
```
digest(5, 'md5');
digest(dom.load('5'), 'sha1');
```

The implementation of this function would take any steps necessary to convert `value` to a `Reader`, then use existing ion-hash-js functionality to compute and return the hash.

### PR DESCRIPTION
**Pull request by pbcornell**
Resolves #43 

* adds and exports `digest(value: any, algorithm: string): Uint8Array` function
* adds unit tests
* incorporates `digest()` into IonHashTests and BigListOfNaughtyStrings test suites

Build currently fails becuase ion-hash-js depends on a version of ion-js that doesn't export `dom`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/DigestTest.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (zslayton) [inline tests/DigestTest.ts]: I believe current guidance is that we no longer need to specify a year. If you prefer to add the year, however, it should be 2020 for a new file.

File: tests/IonHashTests.ts [hunk 2]
```diff
@@ -84,12 +85,19 @@ function writerDigester(writer: Writer, ionData: string | Uint8Array, algorithm:
     hashWriter.digest();
 }
 
+let digestFunctionDigester: Digester = (ionData: string | Uint8Array, algorithm: string, hasherLog: string[]): void => {
```
Comment (zslayton) [inline tests/IonHashTests.ts]: Nit: this line is a bit long.

File: src/IonHash.ts [hunk 3]
```diff
@@ -121,3 +121,24 @@ export function cryptoHasherProvider(algorithm: string): HasherProvider {
     return (): Hasher => { return new _CryptoHasher(algorithm) };
 }
 
+/**
+ * Computes the Ion hash for the given a value and the name of an algorithm known to
```
[the remark this answers is not in the document]
Comment (pbcornell) [reply src/IonHash.ts]: reworded to fix grammar and (hopefully) improve clarity

File: tests/DigestTest.ts (hunk 1 above)
Comment (pbcornell) [reply tests/DigestTest.ts]: removed year, as it is no longer necessary

File: tests/IonHashTests.ts (hunk 2 above)
Comment (pbcornell) [reply tests/IonHashTests.ts]: split this line as well as several others of similar length

**Pull request merged**